src/TDB2.cpp

```diff
@@ -355,23 +355,6 @@ void TF2::load_gc (Task& task)
   {
     Context::getContext ().tdb2.pending._tasks.push_back (task);
   }
-  else if (status == "waiting")
-  {
-    Datetime wait (task.get_date ("wait"));
-    if (wait < now)
-    {
-      task.set ("status", "pending");
-      task.remove ("wait");
-      // Unwaiting pending tasks is the only case not caught by the size()
-      // checks in TDB2::gc(), so we need to signal it here.
-      Context::getContext ().tdb2.pending._dirty = true;
-
-      if (Context::getContext ().verbose ("unwait"))
-        Context::getContext ().footnote (format ("Un-waiting task {1} '{2}'", task.id, task.get ("description")));
-    }
-
-    Context::getContext ().tdb2.pending._tasks.push_back (task);
-  }
```

Comment:
By completely removing this, the tasks that were already in the `status:waiting` state before the upgrade (expired or not) will be moved to `completed.data`:

```
[warrior@pr-2563 ~]$ rm -rf ~/.task
[warrior@pr-2563 ~]$ task add Actionable task wait:yesterday
Created task 1.
[warrior@pr-2563 ~]$ sed -i 's/pending/waiting/' ~/.task/pending.data 
[warrior@pr-2563 ~]$ task all

ID St UUID     Age  Description    
 1 P  9b38093e 1min Actionable task

1 task
[warrior@pr-2563 ~]$ cat ~/.task/pending.data 
[warrior@pr-2563 ~]$ cat ~/.task/completed.data 
[description:"Actionable task" entry:"1628393350" modified:"1628393350" status:"waiting" uuid:"9b38093e-3eef-42a0-84d3-7c811595ffbe" wait:"1628294400"]
```

We need a slight upgrade procedure here. I'll implement this.

Reply:
Good catch :)